services/src/main/java/org/apache/druid/server/AsyncQueryForwardingServlet.java

```diff
@@ -350,10 +369,18 @@ private void handleQueryParseException(
     // Write to the response
     response.setStatus(HttpServletResponse.SC_BAD_REQUEST);
     response.setContentType(MediaType.APPLICATION_JSON);
-    objectMapper.writeValue(
-        response.getOutputStream(),
-        ImmutableMap.of("error", errorMessage)
-    );
+    if (serverConfig.isFilterResponse()
+        && serverConfig.getResponseWhitelistRegex().stream().noneMatch(pattern -> pattern.matcher(errorMessage).matches())) {
+      objectMapper.writeValue(
+          response.getOutputStream(),
+          ImmutableMap.of("error", DEFAULT_QUERY_PARSE_EXCEPTION_MESSAGE)
```

Comment:
Maybe it is finally time to make a basic `ErrorResponse` class to capture this `ImmutableMap.of("error", ...)` pattern, searching the code-base I see it in a lot of places...

On the other hand, it is somewhat inconsistent with `QueryException` though, which uses `error` for the errorCode and has separate message, class, and some additional information. Since this is specifically forwarding queries, I wonder if this should be a type of `QueryException`? (Especially if we bake a method into `QueryException` to make them clean themselves up if configured as such)

Reply:
The response format here is inconsistent with `QueryException`. I am not sure why this was done this way. However, in this PR my intention is to keep the response format the same as original. Notice, that this PR does not add any new field in the  response format, only remove fields that could already be null, and change string message for existing fields. 

However, both Exception thrown here and the Exception from further downstream which uses `QueryException` already make the response inconsistent for the same API call. Since, the format of `QueryException` is well documented, I think we should change this to use `QueryException` too.